mantidproject/mantid
# Set parameter from log value in multi dataset fitting interface

Added a feature requested by scientists: in the "edit local parameter values" dialog, add options to set parameter values from a sample log.

The value can be set for one spectrum, or for all spectra. The log used can be chosen via a drop-down list, and, if it is a time series, there is the choice of taking the mean, min, max, first or last value.

Also noticed and fixed another problem: the "paste" function pasted data into the wrong column. 

**To test:**
- Load some workspaces that contain sample logs (or use the script in the comment below).
- Open _Interfaces/General/Multi dataset fitting_
- Add workspaces with "Add workspaces" button
- Set up a fit function
- Click on a non-global parameter. A small button will appear, use this to get the _Edit local parameter values_ dialog:

![mdfbutton](https://cloud.githubusercontent.com/assets/15363125/17253613/651cc14e-55a9-11e6-90d0-3c46e1e751e8.PNG)

1) _"Set to log" options_
- In the dialog, open the drop-down menu next to one of the parameter values. Confirm new options are present to "set to log" and "set all to log", but they are not yet enabled.
- Tick the checkbox next to "Log" at the bottom and verify the options are now enabled.
- In the log combo box, select a log that is either a numeric value or a numeric time series.
- Try "Set to log" on one of the parameters. The value should be set to the value of the chosen sample log in that workspace.
- Try "Set all to log". All spectra should have their parameter values set to the sample log value from their respective workspaces.
- (If you chose a non-numeric log, you will get an exception, likewise if you selected a log that isn't shared by all workspaces. The user is required to select a sensible option...)
  ![mdfsettolog](https://cloud.githubusercontent.com/assets/15363125/17253870/7b999540-55aa-11e6-9e98-c57d197bfff8.png)

2) _Paste values into table_
- Copy a list of numbers (same length as number of spectra) to the clipboard. They can be separated by newlines or commas. For example, if you loaded 3 workspaces, you could copy "1, 2, 3"
- Right-click on the header of the "Value" column and select "Paste"
- The values should be pasted into the "Value" column (and not the "Role" column)

Fixes #16494.

[Release notes](https://github.com/mantidproject/mantid/blob/d624e14c6474b1d4164035bc41dda2002e458e03/docs/source/release/v3.8.0/ui.rst#multi-dataset-fitting-interface-improvements) have been updated, with an image. There is no user documentation/tests for this interface.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
